tool/tsh/common/mcp.go

```diff
@@ -0,0 +1,232 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package common
+
+import (
+	"context"
+	"log/slog"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/client/proto"
+	apidefaults "github.com/gravitational/teleport/api/defaults"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/client"
+	dbmcp "github.com/gravitational/teleport/lib/client/db/mcp"
+	pgmcp "github.com/gravitational/teleport/lib/client/db/postgres/mcp"
+	"github.com/gravitational/teleport/lib/defaults"
+	"github.com/gravitational/teleport/lib/srv/alpnproxy"
+	"github.com/gravitational/teleport/lib/tlsca"
+	"github.com/gravitational/teleport/lib/utils/listener"
+)
+
+type mcpCommands struct {
+	db *mcpDBCommand
+}
+
+func newMCPCommands(app *kingpin.Application) *mcpCommands {
+	mcp := app.Command("mcp", "View and control proxied MCP servers.")
+	return &mcpCommands{
+		db: newMCPDBCommand(mcp),
+	}
+}
+
+// mcpDBCommand implements `tsh mcp db` command.
+type mcpDBCommand struct {
+	*kingpin.CmdClause
+
+	databaseUser        string
+	databaseName        string
+	labels              string
+	predicateExpression string
+}
+
+func newMCPDBCommand(parent *kingpin.CmdClause) *mcpDBCommand {
+	cmd := &mcpDBCommand{
+		CmdClause: parent.Command("db", "Start a local MCP server for database access"),
+	}
+
+	cmd.Flag("db-user", "Database user to log in as.").Short('u').StringVar(&cmd.databaseUser)
+	cmd.Flag("db-name", "Database name to log in to.").Short('n').StringVar(&cmd.databaseName)
+	cmd.Flag("labels", labelHelp).StringVar(&cmd.labels)
+	cmd.Flag("query", queryHelp).StringVar(&cmd.predicateExpression)
+	return cmd
+}
+
+func (c *mcpDBCommand) run(cf *CLIConf) error {
+	logger, err := initMCPLogger(cf)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	registry := defaultDBMCPRegistry
+	if cf.databaseMCPRegistryOverride != nil {
+		registry = cf.databaseMCPRegistryOverride
+	}
+
+	// Set the labels so it gets parsed when the client is created.
+	cf.Labels = c.labels
```

Comment:
What i meant was if you pass cf.Labels and cf.PredicateExpression to `tc`, you can just use `tc.ResourceFilter(types.KindDatabaseServer)` instead of manually defining the filter. really up to you if you want to use that or manually define the filter. sorry for the confusion